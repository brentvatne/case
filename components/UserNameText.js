import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'

import NavigatorService from '../utilities/navigationService'

import layout from '../constants/Layout'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingLeft: layout.padding,
  },
  text: {
    fontWeight: 'bold',
  },
})

export default class UserNameText extends React.Component {
  constructor(props) {
    super(props)
    this.goToProfile = this.goToProfile.bind(this)
  }

  goToProfile() {
    NavigatorService.navigate('profile', { id: this.props.user.slug })
  }

  render() {
    return (
      <TouchableOpacity onPress={this.goToProfile}>
        <Text
          style={styles.text}
          onPress={this.goToProfile}
        >
          {this.props.user.name}
        </Text>
      </TouchableOpacity>
    )
  }
}

UserNameText.propTypes = {
  user: PropTypes.shape({
    name: PropTypes.string,
    slug: PropTypes.string,
  }).isRequired,
}
