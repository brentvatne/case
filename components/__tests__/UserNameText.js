import 'react-native'

import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import UserNameText from '../UserNameText'

const user = {
  name: 'Damon Zucconi',
  slug: 'damon-zucconi',
}

test('renders correctly', () => {
  const userName = shallow(
    <UserNameText user={user} />,
  )
  expect(toJSON(userName)).toMatchSnapshot()
})
