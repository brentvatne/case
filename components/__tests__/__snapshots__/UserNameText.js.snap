// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Damon Zucconi
     
  </Text>
</TouchableOpacity>
`;
